project: introduce and use pull progress reporter

ffromani opened the pull request:
This patch introduces the progress report interface for pull
operations, alongside with the needed changes to the logging
infrastructure:

1. learn and report if stderr is a tty (thus interactive) or not
2. change the default logging format depending on being interactive
3. use nicer download progress report when interactive.
   unfortunately the podman API doesn't tell (yet?) the data
   being transferred nor the total to be transferred
4. use more frequent reporting when interactive, less frequent
   when running in batch mode
5. vendor the needed deps.

Resolves: https://github.com/fromanirh/pack8s/issues/25
Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
project: introduce and use pull progress reporter

This patch introduces the progress report interface for pull
operations, alongside with the needed changes to the logging
infrastructure:

1. learn and report if stderr is a tty (thus interactive) or not
2. change the default logging format depending on being interactive
3. use nicer download progress report when interactive.
   unfortunately the podman API doesn't tell (yet?) the data
   being transferred nor the total to be transferred
4. use more frequent reporting when interactive, less frequent
   when running in batch mode
5. vendor the needed deps.

Resolves: https://github.com/fromanirh/pack8s/issues/25
Signed-off-by: Francesco Romani <fromani@redhat.com>

fromanirh commented:
/cc @ksimon1

ksimon1 commented on cmd/show.go:
```diff
@@ -36,8 +33,7 @@ func showContainers(cmd *cobra.Command, args []string) error {
 
-	ctx := context.Background()
-	hnd, err := podman.NewHandle(ctx, cOpts.PodmanSocket, cmdutil.NewLogger(cOpts.Verbose))
+	hnd, _, err := cOpts.GetHandle()
```
What about logging here? In this file, there are multiple usage of fmt.Printf. Do we want to keep it that way (same usage in cmd/ports.go)?

ffromani replied:
good point, clarified in a comment

ffromani committed:
show: clarified the reasons we DO NOT use log

The show command is the only place we *want* to write
on stdout, and not on logger!

Signed-off-by: Francesco Romani <fromani@redhat.com>

ksimon1 reviewed (approved):
LGTM
